### ISSUE TITLE
price_unit on Invoice line is Monetary field and ignores precision of product price.

### ISSUE THREAD

**Issue by kmetaxas** [odoo/odoo#9500]
Field `price_unit` of `AccountInvoiceLine` is a `Monetary` field which inherits precision from `Currency`.

**Problem**: We have a 5 decimal digit precision on product price and cost. We get invoice with 5 decimal digits unit price but when we try to register invoice (vendor bills), our decimal digits are rounded to 2 (or whatever precision relevant `Currency` has).

This is wrong. We get rounding errors and wrong total on invoice. Changing `Currency` precision is also wrong (produces wrong rounding on totals, VAT etc.

My solution ( addons/account/models/account_invoice.py ):

Replace:

```python
price_unit = fields.Monetary(string='Unit Price', required=True)
```

With

```python
    price_unit = fields.Float(string="Unit Price",
            digits=dp.get_precision('Product Price'),
            required=True)
```

**Comment by JayVora-SerpentCS** [comment]
@kmetaxas https://youtu.be/hRr2thfZQgc would explain!

**Comment by kmetaxas** [comment]
@JayVora-SerpentCS Changing the accuracy on the currency is the wrong approach. Invoices and accounting in Greece (and i assume many EU countries) are restricted to 2 decimal places (in Greece it is **by law**). 

The law permits more decimal places ( i believe up to 5) for unit price of very price sensitive items (ie selling a low priced item by the tens of thousands) but it will **still** have to be rounded to 2 decimal places **per line total**.  So using a Monetary (Currency) is basically the wrong field for AccountInvoiceLine.price_unit as changing it would also change invoice totals etc etc.

**Comment by SerpentCS** [comment]
Sure, if the localisation does not permit, one can always keep separate DPs for each segment.

Odoo gives a way for it.

**Comment by ferdiga** [comment]
we implemented a possible solution in 6.1
http://bazaar.launchpad.net/~camptocamp/c2c-rd-addons/6.1/view/head:/c2c_product_price_unit/__openerp__.py
reason, some but not all products are invoiced with prices in per 100 or per 1000 units
you do not want to have a purchase price for cars with 5 digits just because you sell screws wit a few euro/cents per 1000

**Comment by kobra34** [comment]
İts very good. Thank you kmetaxas

**Comment by kmetaxas** [comment]
This bug was fixed in commit 2e4777bb3288b8c56b2e747d07c0c2d049fa007e
Closing.

**Comment by tkontogi** [comment]
The problem also exists in 8?

Solution?
